presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxPeriodicSampler.java

```diff
@@ -69,8 +70,10 @@ public JmxPeriodicSampler(
     @PostConstruct
     public void start()
     {
-        lastDumpTimestamp = roundToPeriod(currentTimeMillis());
-        schedule();
+        if (tableHandles.size() > 0) {
```

Comment:
Maybe you would like to create `newSingleThreadScheduledExecutor` in this if branch or something like that? Because now indeed nothing will be scheduled, but there will be an idle thread?